### PR TITLE
Add support for pulling topic name from message

### PR DESCRIPTION
**Pull request by jcf**
This makes it possible to override which topic is used when writing
messages to Kafka.

Simply include the topic name alongside your message, like so:

``` clj
{:message "Hello Kafka!"
 :topic "some.other.topic"}
```

### CODE REVIEW

File: src/onyx/plugin/kafka.clj [hunk 1]
```diff
@@ -324,10 +324,13 @@
...
-                                 p (some-> m :partition int)]
-                             (assert k-message
-                                     "Messages must be supplied in a map in form {:message :somevalue}, with optional :key and :partition keys.")
-                             (send-async! producer (ProducerRecord. topic p k-key (serializer-fn k-message)))))
+                                 p (some-> m :partition int)
+                                 message-topic (get m :topic topic)]
+                             (assert
+                              k-message
+                              (str "Payload missing required :message! Got "
+                                   (pr-str m)))
```
Comment (jcf) [inline src/onyx/plugin/kafka.clj]: I assume printing the object won't cause problems here, but maybe with very large messages and certain tools…?

File: test/onyx/plugin/output_test.clj [hunk 2]
```diff
@@ -63,5 +67,9 @@
                             (select-keys msg [:key :value :partition]))))
                [{:key 1 :value {:n 0} :partition 0}
                 {:key nil :value {:n 1} :partition 0}
-                {:key "tarein" :value {:n 2} :partition 0}])))
+                {:key "tarein" :value {:n 2} :partition 0}]))
+        (is (= [{:key nil :value {:n 3} :partition 0 :topic "message-specific"}]
+               (take-until-done
+                zk-address other-test-topic
+                (fn [v] (read-string (String. v "UTF-8")))))))
```
Comment (jcf) [inline test/onyx/plugin/output_test.clj]: I'm having trouble running tests locally. I'm hoping CI will do a better job than my localhost.

File: src/onyx/plugin/kafka.clj (hunk 1 above)
Comment (lbradstreet) [inline src/onyx/plugin/kafka.clj]: It's possible, but I think the main thing is that the assert will still cause an exception.

**Comment by lbradstreet** [comment]
I think you'll need to loosen the schema to make topic optional in here: https://github.com/onyx-platform/onyx-kafka/blob/0.9.x/src/onyx/tasks/kafka.clj

**Comment by jcf** [comment]
> I think you'll need to loosen the schema to make topic optional in here: https://github.com/onyx-platform/onyx-kafka/blob/0.9.x/src/onyx/tasks/kafka.clj

@lbradstreet I wondered about that.

If I relax that schema we could end up with message payloads that don't have a `:topic`, and then we have messages that can't be written to Kafka. I can add an assertion to make sure we always have a `:topic` in the message payload if the `:kafka/topic` is nil…?

**Comment by jcf** [comment]
New logging shows we're creating our topics: https://gist.github.com/jcf/848823ee14cccbc3ddf2a6200f3c66a0#file-test-log-L29

**Comment by lbradstreet** [comment]
Fantastic. Thank you!

**Pull request merged**